docs/troubleshoot/saml.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: default
+title: Troubleshoot SAML
+parent: Troubleshoot
+nav_order: 2
+---
+
+# SAML troubleshooting
+
+This page includes troubleshooting steps for using SAML for Kibana authentication.
+
+
+---
+
+#### Table of contents
+- TOC
+{:toc}
+
+
+---
+
+## Check sp.entity_id
+
+Most identity providers (IdPs) allow you to configure multiple authentication methods for different applications. For example, in Okta, these clients are called "Applications." In Keycloak, they are called "Clients." Each one has its own entity ID. Make sure to configure `sp.entity_id` to match those settings:
+
+```yml
+saml:
+  ...
+  http_authenticator:
+    type: 'saml'
+    challenge: true
+    config:
+      ...
+      sp:
+        entity_id: kibana-saml
+```
+
+
+## Check the SAML assertion consumer service URL
+
+After a successful login, your IdP sends a SAML response using HTTP POST to Kibana's "assertion consumer service URL" (ACS).
+
+The endpoint the Kibana Security plugin provides is:
+
+```
+/_opendistro/_security/saml/acs
+```
+
+Make sure that you have configured this endpoint correctly in your IdP. Some IdPs also require you to whitelist all endpoints that they send requests to. Ensure that the ACS endpoint is listed.
+
+Kibana also requires you to whitelist this endpoint. Make sure you have the following entry in `kibana.yml`:
+
+```
+server.xsrf.whitelist: [/_opendistro/_security/saml/acs]
+```
+
+
+## Sign all documents
+
+Some IdPs do not sign the SAML documents by default. Make sure the IdP signs all documents.
+
+
+#### Keycloak
+
+![Keycloak UI](../../images/saml-keycloak-sign-documents.png)
+
+
+## Role settings
+
+Including user roles in the SAML response is dependent on your IdP. For example, in Keycloak, this setting is in the **Mappers** section of your client. In Okta, you have to set group attribute statements. Make sure this is configured correctly and that the `roles_key` in the SAML configuration matches the role name in the SAML response:
+
+```yml
+saml:
+  ...
+  http_authenticator:
+    type: 'saml'
+    challenge: true
+    config:
+      ...
+      roles_key: Role
+```
+
+
+## Inspect the SAML response
+
+If you are not sure what the SAML response of your IdP contains and where it places the username and roles, you can enable debug mode in the `log4j2.properties`:
+
+```
+logger.token.name = com.amazon.dlic.auth.http.saml.Token
+logger.token.level = debug
+```
```

Comment:
Hey @hardik-k-shah, I could use some help on the correct strings for these logging settings. I wasn't able to dig them up in code.

Reply:
Looks good.